src/internal/dots.c

```diff
@@ -452,6 +452,7 @@ static sexp* dots_unquote(sexp* dots, struct dots_capture_info* capture_info) {
       r_abort("Internal error: `OP_DOTS_MAX`");
     }
 
+    r_mark_shared(expr);
```

Comment:
I'm unsure this is the right place.

Reply:
It's also possible that on R 4.0.0 we shouldn't be marking as mutable (which sets the refcnt to its maximum value) but we should instead just be incrementing the refcnt by 1.